package/gluon-mesh-batman-adv-core/src/respondd.c

```diff
@@ -212,11 +212,14 @@ static void add_gateway(struct json_object *obj) {
 
 	while (getline(&line, &len, f) >= 0) {
 		char addr[18];
+		unsigned char metric;
+		char nexthop[18];
 
-		if (sscanf(line, "=> %17[0-9a-fA-F:]", addr) != 1)
+		if (sscanf(line, "=> %17[0-9a-fA-F:] (%hhu) %17[0-9a-fA-F:]", addr, &metric, nexthop) != 3) {
```

Comment:
You can just use `%*u` instead of `%hhu` to discard the match, so you don't need the metric variable.